src/features/Chat/chatPageApi.ts

```diff
@@ -161,6 +177,24 @@ export const chatApi = createApi({
         method: 'put',
         body: { status },
       }),
+      async onQueryStarted({ status }, { queryFulfilled }) {
+        try {
+          const statusUpdateSuccessMessages = {
+            ok: 'chat:notification.restoringSuccess',
+            banned: 'chat:notification.banningSuccess',
+            archived: 'chat:notification.archivingSuccess',
+          } as const;
+          await queryFulfilled;
+          toast.success(t(statusUpdateSuccessMessages[status]));
+        } catch (err) {
+          const statusUpdateErrorMessages = {
+            ok: 'chat:notification.restoringError',
+            banned: 'chat:notification.banningError',
+            archived: 'chat:notification.archivingError',
+          } as const;
+          toast.error(t(statusUpdateErrorMessages[status]));
+        }
+      },
```

Comment:
I would move mappings out of try catch just to see the critical flow better.

await in the middle hurt me :D

Reply:
yeah i get it. i was also thinking about it. but in the other hand I like that all of the relevant things are next to each other.